packages/core/src/Utils.ts

```diff
@@ -266,6 +269,20 @@ export class Utils {
       a[j + 1] = element;
     }
   }
+
+  private static encodePathComponents(path: string): string {
+    return path
+      .split("/")
+      .map((component) => {
+        // Retain path structure symbols (., .., empty segments)
+        if (component === "." || component === ".." || component === "") {
+          return component;
+        }
```

Comment:
Is it possible to remove this condition because `encodeURIComponent` will not encode those characters `.` by default?